### PR TITLE
Feature/ersc/rdphoen 897 add swupdate service

### PR DESCRIPTION
**Pull request by iris-ersc**
Include swupdate in rootfs of imx8mp

Change execution of firmwarezip class. It is now based on UPDATE_PROCEDURE variable.

Needs: https://github.com/iris-GmbH/iris-kas/pull/82

### CODE REVIEW

File: conf/multiconfig/qemuarm64-r1.conf [hunk 1]
```diff
@@ -4,8 +4,6 @@
 MACHINE = "qemuarm64"
 PREFERRED_PROVIDER_virtual/bootloader = "u-boot"
 UBOOT_MACHINE ??= "qemu:arm64_defconfig"
-# skip firmware zip step, as this is a local development image
```
Comment (DanielBaumgart) [inline conf/multiconfig/qemuarm64-r1.conf:7]: Why are these deleted?
QEMU-Images shouldn't be zipped anyway. But now they might get zipped, or am i'm wrong?

File: conf/multiconfig/qemuarm64-r2.conf [hunk 2]
```diff
@@ -4,8 +4,6 @@
 MACHINE = "qemuarm64"
 PREFERRED_PROVIDER_virtual/bootloader = "u-boot"
 UBOOT_MACHINE ??= "qemu:arm64_defconfig"
-# skip firmware zip step, as this is a local development image
```
Comment (DanielBaumgart) [inline conf/multiconfig/qemuarm64-r2.conf:7]: Why are these deleted?
QEMU-Images shouldn't be zipped anyway. But now they might get zipped, or am i'm wrong?

File: conf/multiconfig/qemux86-64-r1.conf [hunk 3]
```diff
@@ -3,8 +3,6 @@
 
 MACHINE = "qemux86-64"
 PREFERRED_PROVIDER_virtual/bootloader = "u-boot"
-# skip firmware zip step, as this is a local development image
```
Comment (DanielBaumgart) [inline conf/multiconfig/qemux86-64-r1.conf:6]: Why are these deleted?
QEMU-Images shouldn't be zipped anyway. But now they might get zipped, or am i'm wrong?

File: conf/multiconfig/qemux86-64-r2.conf [hunk 4]
```diff
@@ -3,8 +3,6 @@
 
 MACHINE = "qemux86-64"
 PREFERRED_PROVIDER_virtual/bootloader = "u-boot"
-# skip firmware zip step, as this is a local development image
```
Comment (DanielBaumgart) [inline conf/multiconfig/qemux86-64-r2.conf:6]: Why are these deleted?
QEMU-Images shouldn't be zipped anyway. But now they might get zipped, or am i'm wrong?

File: recipes-support/swupdate/swupdate_%.bbappend [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+
+FILESEXTRAPATHS_prepend := "${THISDIR}/${PN}:"
+SRC_URI_append := "file://defconfig"
+
+FILES_${PN} += "${SWUPDATE_HW_COMPATIBILITY_FILE}"
+
+do_install_append () {
+	install -d $(basename -- ${D}${SWUPDATE_HW_COMPATIBILITY_FILE})
+
+	# all machines are v1.0 for now
```
Comment (DanielBaumgart) [inline recipes-support/swupdate/swupdate_%.bbappend:12]: This needs to change if we change Hardware, right? Replace Storage-Chips and such stuff?

**Review by DanielBaumgart — COMMENTED** [review]
Some question open, regarding QEMU-Images and zipping those firmwares.

If im'm correct they now get zipped, even if this isn't needed as use-case.

File: classes/irma6-firmware-zip.bbclass [hunk 6]
```diff
@@ -29,6 +29,9 @@ python do_createfirmwarezip() {
     import errno
     import tempfile
 
+    if "firmwarezip" not in d.getVar('UPDATE_PROCEDURE', ''):
+        return
+
     if d.getVar('SKIP_FIRMWARE_ZIP', True) == "1":
```
Comment (Jasper-Ben) [inline classes/irma6-firmware-zip.bbclass:35]: when introducing the "UPDATE_PROCEDURE" variable (which is a good idea IMO), why not delete the "SKIP_FIRMWARE_ZIP" logic?

File: conf/multiconfig/qemuarm64-r1.conf (hunk 1 above)
Comment (Jasper-Ben) [reply conf/multiconfig/qemuarm64-r1.conf:7]: this is not needed anymore after introducing the `UPDATE_PROCEDURE` variable in `classes/irma6-firmware-zip.bbclass`. Now images are only zipped, if their config contains the string "firmwarezip" within `UPDATE_PROCEDURE`

**Review by Jasper-Ben — CHANGES_REQUESTED** [review]

**Comment by Jasper-Ben** [comment]
This currently breaks:

```
[AWS CodeBuild Plugin] ERROR: mc:qemux86-64-r1:irma6-test-2.0.2-dev-r0 do_createfirmwarezip: Error executing a python function in exec_func_python() autogenerated:

[AWS CodeBuild Plugin] 

[AWS CodeBuild Plugin] The stack trace of python calls that resulted in this exception/failure was:

[AWS CodeBuild Plugin] File: 'exec_func_python() autogenerated', lineno: 2, function: <module>

[AWS CodeBuild Plugin]      0001:

[AWS CodeBuild Plugin]  *** 0002:do_createfirmwarezip(d)

[AWS CodeBuild Plugin]      0003:

[AWS CodeBuild Plugin] File: '/codebuild/output/src851991482/src/meta-iris-base/classes/irma6-firmware-zip.bbclass', lineno: 32, function: do_createfirmwarezip

[AWS CodeBuild Plugin]      0028:    import yaml

[AWS CodeBuild Plugin]      0029:    import errno

[AWS CodeBuild Plugin]      0030:    import tempfile

[AWS CodeBuild Plugin]      0031:

[AWS CodeBuild Plugin]  *** 0032:    if "firmwarezip" not in d.getVar('UPDATE_PROCEDURE', ''):

[AWS CodeBuild Plugin]      0033:        return

[AWS CodeBuild Plugin]      0034:

[AWS CodeBuild Plugin]      0035:    if d.getVar('SKIP_FIRMWARE_ZIP', True) == "1":

[AWS CodeBuild Plugin]      0036:        return

[AWS CodeBuild Plugin] Exception: TypeError: argument of type 'NoneType' is not iterable

[AWS CodeBuild Plugin] 
```

see Jenkins run https://jenkins.devops.iris-sensing.net/blue/organizations/jenkins/iris-kas/detail/PR-82/2/pipeline/154

File: recipes-support/swupdate/swupdate_%.bbappend (hunk 5 above)
Comment (iris-ersc) [reply recipes-support/swupdate/swupdate_%.bbappend:12]: Exactly. Maybe it is better to write this file elsewhere, during the initial operation in production e.g.

But right now, it is at least a reminder, that this file is necessary later on.

**Comment by iris-ersc** [comment]
Image building for qemu, sc573 and imx8mp are now tested

File: classes/irma6-firmware-zip.bbclass (hunk 6 above)
Comment (iris-ersc) [reply classes/irma6-firmware-zip.bbclass:35]: I have another branch that moves the firmwarezip logic in its own recipe.
For this PR I wanted to touch as little as possible in this bbclass.

**Review by Jasper-Ben — APPROVED** [review]
Jenkins builds now. LGTM

**Pull request merged**